src/media/js/addon/components/addon.js

```diff
@@ -109,15 +119,112 @@ export class Addon extends React.Component {
               <dd>{this.props.latest_version.version}</dd>
             </di>
           }
+
+          {this.props.description &&
+            <di>
+              <dt>Description</dt>
+              <dd>{this.props.description}</dd>
+            </di>
+          }
+
+          {this.props.status &&
+            <di>
+              <dt>Status</dt>
+              <dd>
+                {constants.humanizeAddonStatus(this.props.status,
+                                               this.props.disabled)}
+              </dd>
+            </di>
+          }
+
+          {this.props.latest_version &&
+            <di>
+              <dt>Last version uploaded</dt>
+              <dd>{this.props.latest_version.created}</dd>
+            </di>
+          }
+
+          {marketplaceURL &&
+            <di>
+              <dt>View on Marketplace</dt>
+              <dd>
+                <a href={marketplaceURL}>{marketplaceURL}</a>
+              </dd>
+            </di>
+          }
         </dl>
+      </div>
+    );
+  }
+}
 
-        {this.props.description &&
-          <di>
-            <dt>Description</dt>
-            <dd>{this.props.description}</dd>
+
+export class DashboardAddonListing extends AddonListing {
+  render() {
+    return (
+      <ul className="addon-listing--dashboard">
+        {this.props.addons.map(addon =>
+          <DashboardAddon {...this.props} {...addon}/>
+        )}
+      </ul>
+    );
+  }
+}
+
+
+export class DashboardAddon extends Addon {
+  renderLastUpdated() {
+    if (this.props.latest_version) {
+      const lastUpdated = new Date(this.props.latest_version.created);
+      const months = [
+        "January", "February", "March", "April", "May", "June", "July",
+        "August", "September", "October", "November", "December"
+      ];
+      return (
+        <di>
+          <dt>Updated</dt>
+          <dd>{(months[lastUpdated.getMonth()] + ' ' + lastUpdated.getDate() +
+               ', ' + lastUpdated.getFullYear())}</dd>
+        </di>
+      );
+    }
+  }
+
+  render() {
+    const marketplaceURL = this.marketplaceURL();
+    return (
+      <li className="addon--dashboard">
+        <h2>
+          <ReverseLink to={this.props.linkTo} params={{slug: this.props.slug}}>
+            {this.props.name}
+          </ReverseLink>
+        </h2>
+        <dl className="addon--dashboard--meta">
+          <di className={'addon--status-' + this.props.status}>
```

Comment:
I'm not familiar with the syntax you suggest, and nothing obvious is working like I'd expect. What's the suggestion, precisely?